pkg/rke2/serviceaccount.go

```diff
@@ -0,0 +1,64 @@
+package rke2
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func restrictServiceAccount(ctx context.Context, namespace string, cs *kubernetes.Clientset) error {
+	var sa *v1.ServiceAccount
+	var err error
+	for {
```

Comment:
There are examples of retry logic in this package in the ns.go and psp.go files that would probably be more appropriate or at least more consistent.

Reply:
Moved up a function.